src/geoutils.jl

```diff
@@ -111,3 +111,73 @@ function flipud!(ga::GeoArray)
     ga.f = f
     ga
 end
+
+"""Check bbox overlapping
+
+Return true if two bboxes overlap.
+"""
+function bbox_overlap(
+    bbox_a::NamedTuple{(:min_x, :min_y, :max_x, :max_y)},
+    bbox_b::NamedTuple{(:min_x, :min_y, :max_x, :max_y)})
+    # TODO Define type
+
+    # Check if bboxes are valid
+    if (bbox_a.min_x >= bbox_a.max_x) ||
+        (bbox_a.min_y >= bbox_a.max_y)
+        error("Invalid bbox ", bbox_a)
+    end
+    if (bbox_b.min_x >= bbox_b.max_x) ||
+        (bbox_b.min_y >= bbox_b.max_y)
+        error("Invalid bbox ", bbox_b)
+    end
+
+    # Check if bboxes overlap
+    if (bbox_a.max_x < bbox_b.min_x) ||
+        (bbox_a.max_y < bbox_b.min_y) ||
+        (bbox_a.min_x > bbox_b.max_x) ||
+        (bbox_a.min_y > bbox_b.max_y)
+        return false
+    else
+        return true
+    end
+end
+
+"""
+    function crop(ga::GeoArray, cbox::NamedTuple{(:min_x, :min_y, :max_x, :max_y),Tuple{Float64,Float64,Float64,Float64}})
+
+Crop input GeoArray by coordinates (box or another GeoArray). If the coordinates range is larger than the GeoArray, only the overlapping part is given in the result. 
+"""
+function crop(ga::GeoArray, cbox::NamedTuple{(:min_x, :min_y, :max_x, :max_y)})
+    # TODO Define type
+    # Check if ga and cbox overlap
+    if !bbox_overlap(bbox(ga), cbox)
+        error("GeoArray and crop box do not overlap")
+    end
+    # Check extent and get bbox indices
+    ga_x, ga_y, = size(ga)
+    i_min_x, i_min_y = indices(ga, [cbox.min_x, cbox.max_y])
+    i_max_x, i_max_y = indices(ga, [cbox.max_x, cbox.min_y])
+
+    # bbox is larger
+    i_max_x -= 1
+    i_max_y -= 1
```

Comment:
This shouldn't be required? See #79 for a breaking change for these conversions such as `indices`, which now use the center coordinates.

Reply:
I like what you did in #79. This can be removed when it is merged to master.